helix-term/src/application.rs

```diff
@@ -273,7 +275,16 @@ impl Application {
                     all_threads_stopped,
                     ..
                 }) => {
-                    if let Some(thread_id) = thread_id {
+                    let all_threads_stopped = all_threads_stopped.unwrap_or_default();
+
+                    if all_threads_stopped {
+                        if let Ok(threads) = debugger.threads().await {
+                            for thread in threads {
+                                fetch_stack_trace(debugger, thread.id).await;
+                            }
+                            select_thread_id(&mut self.editor, thread_id.unwrap_or(0), false).await;
```

Comment:
I would drop this select, then change the following "else if" to an "if".

Reply:
Okay, will do after breakpoint verification. Btw we also should set state for all the threads when all threads have been stopped.